src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -813,6 +813,26 @@ void prepare_buffer_fusing::run(program& p) {
                 if (user_info.first) {
                     node.get_users().front()->set_output_layout(user_info.second);
                 }
+
+                // In case that the rank of weight node of gemm is less than 4 and,
+                // it transforms to extend to 4 dims by adding 1 to begin().
+                // Therefore, the padding of crop_layout should be shifted properly.
+                const size_t TDIM = 4;
+                auto user = node.get_users().front();
+                if (user->is_type<gemm>() && pred_layout.is_static() && user->get_dependency(1).id().compare(node.id()) == 0) {
+                    auto input_rank = user->get_kernel_impl_params()->typed_desc<gemm>()->weight_rank;
```

Comment:
Is your target network dynamic & only the gemm node is static? 
Does it use allow_new_shape_infer ?

Reply:
I am curious because adding "1"s to gemm rank < 4 is done by calc_output_layout() I guess, (not calc_output_layouts()). However, the crop in your network has < 4 dim?

Reply:
Added allow_new_shape_infer false condition instead of checking static shape